### ISSUE TITLE
Command line length issues with many module TUs

### ISSUE THREAD

**Issue by kamrann** [build2/build2#420]
Due to `build2` needing to pass module mappings to the compiler, it's relatively easy to hit the compiler command line length limit with MSVC when taking a granular approach to converting a template-heavy codebase to modules (i.e. 1 header -> 1 partition TU or similar). There appears to be a number of conflicting sources of information on the limit and it's possible it varies by shell too, but the limit I seem to be hitting is 32,768 characters. Linux appears to have larger limits, though not I think sufficiently large across the board that this will never become a problem.

There was a discussion on r/cpp a while back which I've searched for now but failed to locate, where there was loose consensus that a granular approach like this is generally not warranted, but I remain unconvinced myself. It seems to me if I want neither many-thousand line interface files, nor to trigger cascading rebuilds when touching anything, then I have no choice other than fine-grained module TUs.

One obvious possibility here would be for `build2` to be able to use compiler support for command files. [MSVC provides this](https://learn.microsoft.com/en-us/cpp/build/reference/cl-command-files?view=msvc-170). I couldn't immediately see an equivalent in Clang, but I did see `-fmodule-map-file` which I am guessing allows to specify multiple module mappings via file rather than as individual arguments.

**Comment by wroyca** [comment]
Related: https://github.com/build2/build2/issues/267

**Comment by boris-kolpackov** [comment]
Yes, we already do this in the link rule. Doing this generally in the compile rule (so that it can be used to handle any kind of repetitive options, such as `-D` and `-I`, not just module mapping) will be complicated by the fact that we call the compiler in multiple places (preprocessing, etc). Will try to get this fixed one way or another for `0.18.0`.
